src/olympia/stats/views.py

```diff
@@ -454,18 +454,31 @@ def contributions_series(request, addon, group, start, end, format):
     qs = (Contribution.objects.extra(select={'date_created': 'date(created)'})
           .filter(addon=addon, amount__gt=0, transaction_id__isnull=False,
                   created__range=date_range)
-          .values('date_created')
-          .annotate(count=Count('amount'), average=Avg('amount'),
-                    total=Sum('amount')))
+          .values('date_created', 'currency')
+          .annotate(count=Count('amount'), total=Sum('amount')))
 
-    # Add `date` and `end` keys for legacy compat.
-    series = sorted(qs, key=lambda x: x['date_created'], reverse=True)
-    for row in series:
-        row['end'] = row['date'] = row.pop('date_created')
+    series = []
+    currencies = set()
+    last = None
+    for row in sorted(qs, key=lambda x: x['date_created'], reverse=True):
```

Comment:
If `qs` is a django `QuerySet` you might want to do `for row in qs.order_by('-date_created'):` to let the database do the work for you.